samples/nrf9160/serial_lte_modem/src/th_util.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#define LOG_DOMAIN util
+#define LOG_LEVEL CONFIG_THIN_LOG_LEVEL
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_DOMAIN);
+
+#include <zephyr.h>
+#include <stdio.h>
+#include <uart.h>
+#include <string.h>
+
+#include <net/socket.h>
+
+/* Partly copy/paste from zephyr/subsys/net/ip/utils.c
+ * Cannot be used when BSD library selects NET_RAW_MODE
```

Comment:
OK, I'll change to use _inet_pton()_

Reply:
Ok, so I've tested it a bit and it seems that `inet_pton` converts the address string correctly. The only difference is that the `inet_pton` returns 1 on success (don't ask me why, IMO it's very confusing, but it's standardized [here](https://pubs.opengroup.org/onlinepubs/9699919799/functions/inet_pton.html)).

So the return value would need some extra attention here (return 0 if `inet_pton` returns 1, and for instance `-EINVAL` otherwise, IMO there's no need for more sophisticated error handling).

Reply:
Actually I got 1 yesterday and thought it fails :)
Now I changed it as you suggested and tested OK.